.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,40 @@
+on:
+    push:
+        branches-ignore:
+            - 'master'
+    pull_request:
+        branches:
+            - 'master'
+jobs:
+    build:
+        runs-on: ubuntu-latest
+        continue-on-error: false
+        steps:
+            - name: Checking out code base
+              uses: actions/checkout@v2
+
+            - name: Set up JDK 1.8
+              uses: actions/setup-java@v1
+              with:
+                java-version: 1.8
+
+            - name: Building package
+              run: mvn clean package -U -DskipTests
+              continue-on-error: false
+
+    test:
```

Comment:
Why build once and then test? both can be done in one shot. It will be much faster